view/frontend/templates/js/replacejs.phtml

```diff
@@ -35,6 +35,62 @@ ob_start();
     // in payment method renderer, vendor/boltpay/bolt-magento2/view/frontend/web/js/view/payment/method-renderer/boltpay.js
     window.boltConfig = <?php echo $block->getSettings(); ?>;
 
+    var trackCallbacks = {
+        onCheckoutStart: function() {
+            try {
+                <?php echo $trackCallbacks['checkout_start']; ?>
+            } catch (error) {
+                console.error(error);
+            }
+        },
+        onEmailEnter: function(email) {
+            try {
+                <?php echo $trackCallbacks['email_enter']; ?>
+            } catch (error) {
+                console.error(error);
+            }
+        },
+        onShippingDetailsComplete: function() {
+            try {
+                <?php echo $trackCallbacks['shipping_details_complete']; ?>
+            } catch (error) {
+                console.error(error);
+            }
+        },
+        onShippingOptionsComplete: function() {
+            try {
+                <?php echo $trackCallbacks['shipping_options_complete']; ?>
+            } catch (error) {
+                console.error(error);
+            }
+        },
+        onPaymentSubmit: function() {
+            try {
+                <?php echo $trackCallbacks['payment_submit']; ?>
+            } catch (error) {
+                console.error(error);
+            }
+        },
+        onSuccess: function() {
+            try {
+                <?php echo $trackCallbacks['success']; ?>
+            } catch (error) {
+                console.error(error);
+            }
+        },
+        onClose: function() {
+            try {
+                <?php echo $trackCallbacks['close']; ?>
+            } catch (error) {
+                console.error(error);
+            }
+        }
+    };
+    var defaultErrorMessage = "<?php echo $block->getBoltPopupErrorMessage() ?>";
+    var successCallback = function(data) {
+        <?php echo $block->getJavascriptSuccess(); ?>
+    };
+
```

Comment:
Moving all the variables above into the global scope might eventually collide with some other piece of code / module. They have pretty common names. Please consider adding them a name prefix or make them properties of some global object (e.g. "Bolt", "Boltpay").

Reply:
done